cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -706,6 +707,32 @@ public void updateAppSpec(ApplicationId appId, ApplicationSpecification spec) th
         GSON.toJson(updated));
   }
 
+  /**
+   * Updates the scm metadata of an application.
+   *
+   * @param appId {@link ApplicationId} of the app
+   * @param scmMeta {@link SourceControlMeta} the new scm metadata (git filehash)
+   * @throws IOException when the update operation fails
+   * @throws ApplicationNotFoundException when the application is not found
+   */
+  public void updateAppScmMeta(ApplicationId appId, SourceControlMeta scmMeta)
+      throws IOException, ApplicationNotFoundException {
+    final StructuredTable appSpecTable = getApplicationSpecificationTable();
+
+    List<Field<?>> fields = getApplicationPrimaryKeys(appId);
+    Optional<StructuredRow> existing = appSpecTable.read(fields);
+    if (!existing.isPresent()) {
+      throw new ApplicationNotFoundException(appId);
+    }
+
+    fields.add(Fields.stringField(StoreDefinition.AppMetadataStore.SOURCE_CONTROL_META, GSON.toJson(scmMeta)));
+    // upsert, because update does not seem to work correctly in the nosql implementation
```

Comment:
what does this mean, what does not work correctly?

It's a big problem if the nosql implementation is broken, would mean anything else using it is broken. If it is indeed broken, please open a blocker jira and reference it in the TODO below

Reply:
Added a blocker JIRA to track this.